### PR TITLE
Introduces NominatimSearch component

### PR DESCRIPTION
**Pull request by KaiVolland**
This introduces the `NominatimSearch` component.

It also adds the `lint:fix` script to the package.json

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,6 +53,7 @@
     "coveralls": "cat coverage/lcov.info | coveralls",
     "test:watch": "karma start karma.conf.js",
     "lint": "eslint --ext js,html,md example-templates/ examples/ src/",
+    "lint:fix": "eslint --fix --ext js,html,md example-templates/ examples/ src/",
```
Comment (marcjansen) [inline package.json]: Or

    npm run lint -- --fix

File: src/Field/NominatimSearch/NominatimSearch.example.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+layout: basic.html
+title: NominatimSearch example
+description: This is a example using an NominatimSearch.
```
Comment (marcjansen) [inline src/Field/NominatimSearch/NominatimSearch.example.md]: a**n** example

File: src/Field/NominatimSearch/NominatimSearch.jsx [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+  AutoComplete,
+} from 'antd';
+const Option = AutoComplete.Option;
+
+import Logger from '../../Util/Logger';
+import UrlUtil from '../../Util/UrlUtil.js';
+import olProj from 'ol/proj';
+
+/**
+ * The ToggleButton.
```
Comment (marcjansen) [inline src/Field/NominatimSearch/NominatimSearch.jsx]: copy paste error

File: src/Field/NominatimSearch/NominatimSearch.jsx [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+ *
+ * @class The NominatimSearch
```
Comment (marcjansen) [inline src/Field/NominatimSearch/NominatimSearch.jsx]: The `The` is found everywhere, but I think it is wrong

File: src/Field/NominatimSearch/NominatimSearch.jsx [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+    map: PropTypes.object.isRequired,
+    minChars: PropTypes.number
+  }
+
+  static defaultProps = {
+    /**
+     * The Nominatim Base URL. See http://wiki.openstreetmap.org/wiki/Nominatim
+     * @type {String}
+     */
+    nominatimBaseUrl: 'http://nominatim.openstreetmap.org/search?',
```
Comment (marcjansen) [inline src/Field/NominatimSearch/NominatimSearch.jsx]: Is there a https variant? If so I'd prefer that.

File: src/Field/NominatimSearch/NominatimSearch.jsx [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+      countrycodes: this.props.countrycodes,
+      q: this.state.searchTerm
+    };
+
+    const getRequestParams = UrlUtil.objectToRequestString(baseParams);
+
+    fetch(`${this.props.nominatimBaseUrl}${getRequestParams}`)
+      .then(response => response.json())
+      .then(this.onFetchSuccess.bind(this))
+      .catch(this.onFetchError);
```
Comment (marcjansen) [inline src/Field/NominatimSearch/NominatimSearch.jsx]: Missing bind?

File: src/Field/NominatimSearch/NominatimSearch.jsx [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+  }
+
+  /**
+   * This function gets called on success of the nomintim fetch.
```
Comment (marcjansen) [inline src/Field/NominatimSearch/NominatimSearch.jsx]: typo nominatim

File: src/Field/NominatimSearch/NominatimSearch.spec.jsx [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+      expect(wrapper.state().dataSource).to.eql(['Peter']);
+    });
+  });
+
+  describe('#onFetchError', () => {
+    it('sets the response as state.dataSource', () => {
+      const wrapper = TestUtil.mountComponent(NominatimSearch);
+      const loggerSpy = sinon.spy(Logger, 'error');
+      wrapper.instance().onFetchError('Peter');
+      expect(loggerSpy.calledWith('Error while requesting Nominatim: Peter')).to.be.ok();
```
Comment (marcjansen) [inline src/Field/NominatimSearch/NominatimSearch.spec.jsx]: would be nice to have calledOn (or similar) for the scope

File: package.json (hunk 1 above)
Comment (KaiVolland) [reply package.json]: Didn't knew that... or forgot it.

File: src/Field/NominatimSearch/NominatimSearch.jsx (hunk 6 above)
Comment (KaiVolland) [reply src/Field/NominatimSearch/NominatimSearch.jsx]: It just logs an error and `this` is not used, so we don't need it. But i'll ad it to be consistent.

File: src/Field/NominatimSearch/NominatimSearch.example.md [hunk 9]
```diff
@@ -0,0 +1,8 @@
+---
+layout: basic.html
+title: NominatimSearch example
+description: This is an example using an NominatimSearch.
```
Comment (dnlkoch) [inline src/Field/NominatimSearch/NominatimSearch.example.md]: …a NominatimSearch.

File: src/Field/NominatimSearch/NominatimSearch.example.md [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+collection: Examples
+---
+
+This demonstrates the use of the NominatimSearch.
```
Comment (dnlkoch) [inline src/Field/NominatimSearch/NominatimSearch.example.md]: …the usage…

File: src/Field/NominatimSearch/NominatimSearch.jsx [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+  onFetchError(error) {
+    Logger.error(`Error while requesting Nominatim: ${error}`);
+  }
+
+  /**
+   * The function describes what to do when an item is selected.
+   *
+   * @param {value} key The key of the selected option.
+   */
+  onMenuItemSelected(key) {
```
Comment (dnlkoch) [inline src/Field/NominatimSearch/NominatimSearch.jsx]: We should introduce a `onSelect` prop to the component, so applications may override this behaviour (e.g. to set the map view via dispatching an action or similiar).

Comment (dnlkoch) [reply src/Field/NominatimSearch/NominatimSearch.jsx]: Ah, never mind. Just noticed that the `AutoComplete` compoent has an `onSelect` that can be used already :duck:

**Pull request merged**